thunder/benchmarks/targets.py

```diff
@@ -812,3 +814,29 @@ def test_resnet50(benchmark, executor: Callable, compute_type: ComputeType):
     fn = executor(b.fn())
 
     benchmark_for_compute_type(compute_type, benchmark, fn, args, kwargs)
+
+
+#
+# Thunder dynamo benchmarks
+#
+
+
+def test_dynamo_LlamaMLPBenchmark(benchmark):
+    backend = ThunderCompilerGraphBenchmarking(benchmark)
+
+    bench: Benchmark = LlamaMLPBenchmark(
+        config="Llama-2-7b-hf",
+        batchdims=(2,),
+        device="cuda:0",
+        dtype=thunder.bfloat16,
+        requires_grad=True,
+    )
+
+    args, kwargs = bench.make_batch()
+    # Using torch.compile here fails with "TypeError: cannot pickle '_io.TextIOWrapper' object" in
+    # https://github.com/Lightning-AI/pytorch-lightning/blob/828fd998961f6a60f92c35254bb94d6e049ad069/src/lightning/fabric/wrappers.py#L421
+    fn = torch._dynamo.optimize(backend=backend)(bench.fn())
```

Comment:
Does `torch.compile` fail here when used instead of `torch._dynamo.optimize`? Do we import lightning package somewhere that replaces real torch.compile with its own?

Reply:
Most likely the whole lightning package is imported in this line:
https://github.com/Lightning-AI/lightning-thunder/blob/5dcbc55a0523a471fdfec26b6b098b1aeb3b1fd8/thunder/benchmarks/targets.py#L11